CRM/Attentively/BAO/Attentively.php

```diff
@@ -0,0 +1,275 @@
+    $response = curl_exec( $ch );
+    $result = get_object_vars(json_decode($response));
+    return $result;
+  }
+
+  static public function pushWatchedTerms($terms) {
+    if (empty($terms)) {
+      return;
+    }
+    $settings = CRM_Core_OptionGroup::values('attentively_auth', TRUE, FALSE, FALSE, NULL, 'name', FALSE);
+    $url = self::checkEnvironment();
+    $url = $url . 'watched_terms_add';
+    $post = 'access_token=' . $settings['access_token'] . '&terms=' . $terms;
+    $ch = curl_init( $url );
+    curl_setopt( $ch, CURLOPT_POST, TRUE);
+    curl_setopt( $ch, CURLOPT_POSTFIELDS, $post);
+    curl_setopt( $ch, CURLOPT_FOLLOWLOCATION, 1);
+    curl_setopt( $ch, CURLOPT_HEADER, 0);
+    curl_setopt( $ch, CURLOPT_RETURNTRANSFER, 1);
+    $response = curl_exec( $ch );
+    $result = get_object_vars(json_decode($response));
+    return $result;
+  }
+
+  static public function getNetworks($cid) {
+    if (!$cid) {
+      return NULL;
+    }
+    $config = CRM_Core_Config::singleton();
+    $sql = "SELECT * FROM civicrm_attentively_member_network
+      WHERE contact_id = {$cid}";
+    $dao = CRM_Core_DAO::executeQuery($sql);
+    while ($dao->fetch()) {
+      if ($dao->name == 'gravatar') {
+        $email = self::getAttentivelyEmail($cid);
+        $network[$dao->name]['url'] = $dao->url;
+        $atts = array(
+          'position' => 'static',
+          'border-top-left-radius' => '5px',
+          'border-top-right-radius' => '5px',            
+        );
+        $network[$dao->name]['image'] = self::getGravatar($email, TRUE, $atts);
+      }
+      else {
+        $network[$dao->name]['url'] = $dao->url;
+        $network[$dao->name]['image'] = '<img class="network-image" src="' .$config->extensionsURL. '/biz.jmaconsulting.attentively/images/' .$dao->name. '.png" style="width:80px !important; height:80px !important;"/>';
+      }
+    }
+    return $network;
+  }
+
+  static public function getKloutScore($cid) {
+    if (!$cid) {
+      return NULL;
+    }
+    $sql = "SELECT klout_score FROM civicrm_attentively_member
+      WHERE contact_id = {$cid}";
+    $klout = CRM_Core_DAO::singleValueQuery($sql);
+    return $klout;
+  }
+
+  static public function getMemberID($cid) {
+    if (!$cid) {
+      return NULL;
+    }
+    $sql = "SELECT member_id FROM civicrm_attentively_member
+      WHERE contact_id = {$cid}";
+    $id = CRM_Core_DAO::singleValueQuery($sql);
+    return $id;
+  }
+
+  static public function getCount($cid) {
+    $sql = "SELECT count(id) FROM civicrm_attentively_member_network
+      WHERE contact_id = {$cid}";
+    $count = CRM_Core_DAO::singleValueQuery($sql);
+    return $count;
+  } 
+  
+  static public function getAttentivelyEmail($cid) {
+    if (!$cid) {
+      return NULL;
+    }
+    $sql = "SELECT email_address FROM civicrm_attentively_member
+      WHERE contact_id = {$cid}";
+    $email = CRM_Core_DAO::singleValueQuery($sql);
+    return $email;
+  }
+
```

Comment:
Can we use single function for getAttentivelyEmail(), getKloutScore(), getMemberID?